runtime/cudaq/cudaq.cpp

```diff
@@ -300,27 +291,31 @@ std::string get_quake_by_name(const std::string &kernelName,
   // Find the quake code
   std::optional<std::string> result;
   std::shared_lock<std::shared_mutex> lock(globalRegistryMutex);
-  for (auto [k, v] : quakeRegistry) {
-    if (k == kernelName) {
-      // Exact match. Return the code.
-      return v;
-    }
-    if (k.starts_with(kernelNamePrefix)) {
-      // Prefix match. Record it and make sure that it is a unique prefix.
-      if (result.has_value()) {
-        if (throwException)
-          throw std::runtime_error("Quake code for '" + kernelName +
-                                   "' has multiple matches.\n");
-      } else {
-        result = v;
-      }
-    }
-  }
+
+  std::find_if(
+      quakeRegistry.begin(), quakeRegistry.end(), [&](const auto &pair) {
+        if (pair.first == kernelName) {
+          result = pair.second;
```

Comment:
I think it would be worth retaining the comment that previously existed here IMO.
```suggestion
          // Exact match. Return the code.
          result = pair.second;
```